firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -362,6 +362,18 @@ public WriteBatch batch() {
     return shutdownInternal();
   }
 
+  /**
+   * Waits until all pending writes that existed at the time of calling have been successfully
+   * written to the server.
+   *
+   * @return A {@code Task} which resolves when all pending writes are written to the backend. If
+   *     there is a Firebase user change, the returned {@code Task} will resolve to an exception. If
+   *     the client is offline, the returned {@code Task} will not resolve.
+   */
+  Task<Void> waitForPendingWrites() {
```

Comment:
I still think we should move this comment to a doc (it could be the API proposal doc) and do some collaborative editing: 

- `pending writes that existed at the time of calling` could be clearer.
- `written to the server` is an unusual way of phrasing the right thing.
- I think we should move the comment on the user change to a separate paragraph and move it out of the `@return` section
- Tasks on Android don't resolve, they complete.

@mikelehen usually has a ton of good feedback on these comments, and his feedback tends to be much more valuable than what I can provide. 

Thanks!

Reply:
The API proposal doc would be a good place to collaborate on this.

Note that the "written to the server" language is used in the iOS documentation already which is why I suggested it: https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Public/FIRDocumentReference.h#L117. I'm definitely open to finding a better way to succinctly describe this.

Reply:
Using Sebastian's version now, looks great.